### PR TITLE
Refactor Additional User Info on Client

### PR DESCRIPTION
**Pull request by anshulg954**
### Change Description

**Brief** (a few bullet points describing your changes, use full sentences and try to link lines in the code whenever needed)
Similar points as described on Server

**Details** (add details if your pull request is more complicated and harder to understand from the code alone)
API hit prevented.

### Standard Qs (leave questions that do not apply blank)
**If you broke behavior: Please describe what behavior you broke and how you inform people to not get stuck trying to use the old behavior.**

**If you used new dependencies: Did you add them to `requirements.txt`?**
na

**Who did you ping on Mattermost to review your PR? Please ping that person again whenever you are ready for another review.**
@SamuelGabriel  

---
*Please do not mark comments/conversations as resolved unless you are the assigned reviewer. This helps maintain clarity during the review process.*

### CODE REVIEW

File: tabpfn_client/client.py [hunk 1]
```diff
@@ -275,8 +277,16 @@ def register(
...
         response = self.httpx_client.post(
             self.server_endpoints.register.path,
-            params={"email": email, "password": password, "password_confirm": password_confirm,
-                    "validation_link": validation_link}
+            params={
+                "email": email, 
+                "password": password, 
+                "password_confirm": password_confirm,
+                "validation_link": validation_link, 
+                "company": additional_info["company"],
```
Comment (SamuelGabriel) [inline tabpfn_client/client.py:285]: i think the better thing to use here is `**additional_info` if you do not have an argument against that

File: tabpfn_client/prompt_agent.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def prompt_and_set_token(cls, user_auth_handler: "UserAuthenticationClient"):
         # Registration
         if choice == "1":
             # validation_link = input(cls.indent("Please enter your secret code: "))
-            validation_link = "tabpfn-2023"
+            validation_link = "tabpfn-test"
```
Comment (SamuelGabriel) [inline tabpfn_client/prompt_agent.py:54]: why is this different now? this won't work in production then, right?

File: tabpfn_client/prompt_agent.py [hunk 3]
```diff
@@ -126,7 +126,12 @@ def prompt_add_user_information(cls, user_auth_handler: "UserAuthenticationClien
         )
         contact_via_email = True if choice_contact == "y" else False
 
-        user_auth_handler.add_user_information(company, role, use_case, contact_via_email)
```
Comment (SamuelGabriel) [inline tabpfn_client/prompt_agent.py:129]: you need to delete UserAuthenticationClient.add_user_information function still, right?

File: tabpfn_client/service_wrapper.py [hunk 4]
```diff
@@ -39,13 +39,11 @@ def set_token_by_registration(
             email: str,
             password: str,
             password_confirm: str,
-            validation_link: str
+            validation_link: str,
+            additional_info: dict
     ) -> tuple[bool, str]:
 
-        is_created, message = self.service_client.register(email, password, password_confirm, validation_link)
-        if is_created:
```
Comment (SamuelGabriel) [inline tabpfn_client/service_wrapper.py:46]: we still do want to set the token here, don't we?

**Review by SamuelGabriel — CHANGES_REQUESTED** [review]
thanks for the PR. i think there are a few things to be done:
i) the cleanup of the old API is not completely done, I think, e.g. in service_wrapper.py
ii) you changed some settings, that I do not think you want to change

File: tabpfn_client/client.py (hunk 1 above)
Comment (anshulg954) [reply tabpfn_client/client.py:285]: Changed to this **additional_info and it is working fine as expected.

File: tabpfn_client/prompt_agent.py (hunk 2 above)
Comment (anshulg954) [reply tabpfn_client/prompt_agent.py]: No changes in this

**Comment by anshulg954** [comment]
Ready for merge

**Review by SamuelGabriel — APPROVED** [review]

**Pull request merged**